Backpack-SwiftUI/Price/Classes/BPKPrice.swift

```diff
@@ -101,13 +107,32 @@ public struct BPKPrice: View {
     
     @ViewBuilder
     private var priceLabel: some View {
-        BPKText(price, style: priceFontStyle)
+        switch alignment {
+        case .leading, .row:
+            BPKText(price, style: priceFontStyle)
+            if icon != nil {
+                redirectingIcon
+                    .offset(y: 2)
+            }
+        case .trailing:
+            HStack(spacing: .sm) {
+                BPKText(price, style: priceFontStyle)
+                if icon != nil {
+                    redirectingIcon
+                }
+            }
+        }
         if let trailingText = trailingText {
             BPKText(trailingText, style: accessoryFontStyle)
                 .foregroundColor(.textSecondaryColor)
         }
     }
     
+    private var redirectingIcon: some View {
```

Comment:
Can you make this a function instead and receive a non optional icon instead of using the ! ?